### PR TITLE
refactor(moderation): remove obsolete evaluator compatibility seams

### PR DESCRIPTION
**Pull request by rmusser01**
## Change summary

This PR removes eight unused private ModerationService forwarding methods now owned by PolicyEvaluator, reducing redundant internal surface without changing supported behavior. Public APIs, dispatch paths, policy compatibility, and moderation semantics remain intact, with direct evaluator tests preserving the relevant boundary cases.

## Summary

- remove eight repository-unused private `ModerationService` evaluator forwarding methods
- keep public facade APIs, active dispatch paths, and `policy_types()` compatibility unchanged
- move unique scan boundary expectations to direct `PolicyEvaluator` coverage

## Test plan

- [x] Python compilation for touched modules and tests
- [x] 299 Moderation unit tests
- [x] 89 Guardian policy tests
- [x] 16 Chat moderation integration tests
- [x] 12 Workflow moderation adapter tests
- [x] Audio transcription redaction integration test
- [x] Ruff and Black checks for changed tests
- [x] Bandit on `moderation_service.py` with zero findings
- [x] PR diff whitespace validation

## Notes

- `moderation_service.py` retains pre-existing Black formatting differences also present on `dev`; this PR does not mass-format the module.
- Removing undocumented private methods may affect unknown external callers; no repository callers remain.

---
## Summary by cubic
Removes eight unused private `ModerationService` forwarding methods that `PolicyEvaluator` now owns, reducing redundant internal surface without changing supported behavior. Public APIs, active dispatch paths, policy compatibility, and moderation semantics remain intact; unique scan boundary expectations moved to direct `PolicyEvaluator` coverage.

**Refactors**
- The deleted methods only forwarded to `PolicyEvaluator` and had no production callers.
- New tests assert the eight names are absent from `ModerationService.__dict__`; obsolete wrapper-specific tests, including descriptor-kind assertions, were removed.
- Removing these undocumented private methods is an intentional callable-surface break; external callers cannot be ruled out.
- `moderation_service.py` keeps pre-existing Black formatting differences also present on `dev`; the module was not mass-formatted.

<sup>Written for commit e65b150e8e83b24bbd92a39ae4bbcb824020b6eb. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/rmusser01/tldw_server/pull/2929?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: tldw_Server_API/tests/unit/test_moderation_policy_evaluator_delegation.py [hunk 1]
```diff
@@ -23,6 +23,31 @@
...
+_OBSOLETE_SCAN_HELPER_DELEGATES = (
+    "_iter_scan_chunks",
+    "_find_match_span",
+    "_collect_rule_matches",
+)
+
+
+def test_obsolete_policy_helper_delegates_are_not_class_local():
+    for name in _OBSOLETE_POLICY_HELPER_DELEGATES:
+        assert name not in ModerationService.__dict__
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/tests/unit/test_moderation_policy_evaluator_delegation.py]: Fixed in e65b150e8e. Removed the two tests that asserted private ModerationService class layout; the direct PolicyEvaluator and public facade behavior coverage remains.

[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/tests/unit/test_moderation_policy_evaluator_delegation.py]: Resolved in e65b150e8e by removing the two private-layout tests, so there are no new undocumented test functions.

File: tldw_Server_API/tests/unit/test_moderation_policy_evaluator_delegation.py [hunk 2]
```diff
@@ -23,6 +23,31 @@
...
+)
+
+_OBSOLETE_SCAN_HELPER_DELEGATES = (
+    "_iter_scan_chunks",
+    "_find_match_span",
+    "_collect_rule_matches",
+)
+
+
+def test_obsolete_policy_helper_delegates_are_not_class_local():
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_Server_API/tests/unit/test_moderation_policy_evaluator_delegation.py]: Resolved in e65b150e8e by removing the two private-layout tests, so there are no new untyped test functions.
